### ISSUE TITLE
nested references

### ISSUE THREAD

**Issue by reesew** [zaggino/z-schema#126]
I'm using the cli, and have been unable to get validation to work when there is more than one level of references. If schema A references schema B, it works fine, but if schema B further references schema C, it fails with either unable to read file schema B, or unable to parse file schema B as json. 
Is this the expected behavior?

### Example:
Given the data 
`data.json`
```
[
    {
        "number": 1
    }
]
```
and using the cli `z-schema ints.json data.json`
where `ints.json` =
```json
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "id": "ints.json#",
    "type": "array",
    "items": {
        "$ref": "int.json#"
    }
}
```
The following schema works
`int.json`
```json
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "id": "int.json#",
    "type": "object",
    "required": [
        "number"
    ],
    "properties": {
        "number": {
            "type": "integer"
        }
    }
}
```
But the following does not work
`int.json`
```json
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "id": "int.json#",
    "type": "object",
    "required": [
        "number"
    ],
    "properties": {
        "number": {
            "$ref": "data-types.json#/definitions/number"
        }
    }
}
```
producing the following stack trace:
```
Error: Couldn't read the file: /path/to/int.json
    at readJson (/usr/local/lib/node_modules/z-schema/bin/z-schema:55:15)
    at /usr/local/lib/node_modules/z-schema/bin/z-schema:116:37
    at Array.forEach (native)
    at validate (/usr/local/lib/node_modules/z-schema/bin/z-schema:91:31)
    at /usr/local/lib/node_modules/z-schema/bin/z-schema:120:25
    at Array.forEach (native)
    at validate (/usr/local/lib/node_modules/z-schema/bin/z-schema:91:31)
    at /usr/local/lib/node_modules/z-schema/bin/z-schema:120:25
    at Array.forEach (native)
    at validate (/usr/local/lib/node_modules/z-schema/bin/z-schema:91:31)
```
Additionally, `z-schema int.json datum.json` works when int.json has the reference and datum.json contains `{"number": 1}`

**Comment by zaggino** [comment]
Thanks for the report, working on the fix.

**Comment by zaggino** [comment]
`z-schema` has been updated to `3.12.1`

**Comment by reesew** [comment]
So for the example posted above, the new error I'm getting is `REMOTE_NOT_VALID` with the message `Remote reference didn't compile successfully: int.json#`, which I'm guessing is the expected behavior given my inputs (i.e. because of my lack of understanding of json schema in general). Do I need to use absolute URIs? Am I using the `#` correctly?

**Comment by zaggino** [comment]
well, the error basically says that in file `int.json` there's a referenced file `data-types.json` which can't be found

**Comment by reesew** [comment]
Right, but why can't it find it when it is the 2nd nested level, but it can when it is only the first nested level (i.e. `z-schema int.json datum.json`) ? It does exist in the same directory as `int.json` and `ints.json`, and has the contents 
```json
{
    "$schema": "http://json-schema.org/draft-04/schema#",
    "id": "data-types.json#",
    "definitions": {
        "number": {
            "type": "integer"
        }
    }
}
```

**Comment by zaggino** [comment]
maybe another bug, let me check this

**Comment by zaggino** [comment]
actually, to make things fast, can you zip contents of your demo and post here the link? or mail them to me to zaggino at gmail

**Comment by reesew** [comment]
Files available here:
https://gist.github.com/reesew/64379e1e1f887a8985af

**Comment by zaggino** [comment]
This is now fixed by `z-schema@3.12.2`

**Comment by reesew** [comment]
amazing
